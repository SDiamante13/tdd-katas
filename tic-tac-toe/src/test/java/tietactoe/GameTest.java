package tietactoe;

import org.approvaltests.Approvals;
import org.approvaltests.StoryBoard;
import org.approvaltests.reporters.QuietReporter;
import org.approvaltests.reporters.UseReporter;
import org.junit.jupiter.api.Test;

@UseReporter(QuietReporter.class)
class GameTest {

    @Test
    void ticTacToeGame() {
        StoryBoard storyBoard = new StoryBoard();
        storyBoard.addDescription("Tic-Tac-Toe");

        TicTacToe ticTacToe = new TicTacToe();
        storyBoard.addFrame("New Game", ticTacToe);

        ticTacToe.place("X", new Coordinate(1, 1));
        storyBoard.addFrame("Player X takes turn", ticTacToe);

        ticTacToe.place("Y", new Coordinate(1, 1));
        storyBoard.addFrame("Player Y makes illegal move", ticTacToe);

        Approvals.verify(storyBoard);
    }
}
