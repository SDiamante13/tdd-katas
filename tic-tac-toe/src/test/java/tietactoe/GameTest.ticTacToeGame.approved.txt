Tic-Tac-Toe


New Game:
+---+---+---+
| _ | _ | _ |
| _ | _ | _ |
| _ | _ | _ |
+---+---+---+


Player X takes turn:
+---+---+---+
| _ | _ | _ |
| _ | X | _ |
| _ | _ | _ |
+---+---+---+


Player Y makes illegal move:
+---+---+---+
| _ | _ | _ |
| _ | X | _ |
| _ | _ | _ |
+---+---+---+
